controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -136,7 +136,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		{spec, "version"},
 		{spec, "upgradeAfter"},
 		{spec, "nodeDrainTimeout"},
-		{spec, "rolloutStrategy"},
+		{spec, "rolloutStrategy", "*"},
```

Comment:
Should we add a test case that ensures the specific problem is fixed?

Reply:
@fabriziopandini added the test that ensure that maxSurge field can be changed. Test also need to upgrade the replica count to be able to change maxSurge value to 0. Do you think this is enough to cover our case here?

Reply:
Might make sense to call .Default() in the tests for create, which would then catch errors created by blocking defaulting

Reply:
> Might make sense to call .Default() in the tests for create, which would then catch errors created by blocking defaulting

@randomvariable could you explain a little more what you meant. Should we do more tests here? https://github.com/kubernetes-sigs/cluster-api/blob/671d2d5b4882924abffe2b7bf7f36924c3baffa3/controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook_test.go#L31

Reply:
> Do you think this is enough to cover our case here?

IMO it does not cover our use case because this test is calling `ValidateDefault`, so it will pass no matter if the current change in `ValidateUpdate` is there or not. In other words, this test is not preventing a regression of this problem.

Reply:
We need a generic test that:

- creates an object with minimum viable information
- calls all the defaulting logic
- calls the validation/admission logic
- succeeds

Often times it's easy to get into a situation where a defaulting webhook interferes with the admission logic because the old vs new are different.

Reply:
I'll do this as a separate PR.

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/4448/files

Reply:
Created #4448